apps/api/src/app/workflows-v2/exceptions/step-has-issues.exception.ts

```diff
@@ -0,0 +1,8 @@
+import { InternalServerErrorException } from '@nestjs/common';
+import { NotificationStepEntity } from '@novu/dal';
+
+export class StepHasIssuesException extends InternalServerErrorException {
```

Comment:
A step with issues should raise a 4XX error, not a 5XX as it's an expected error state based on client provided data.

```suggestion
export class StepHasIssuesException extends InternalServerErrorException {
```